frappe/desk/utils.py

```diff
@@ -76,3 +76,51 @@ def provide_binary_file(filename: str, extension: str, content: bytes) -> None:
 	frappe.response["type"] = "binary"
 	frappe.response["filecontent"] = content
 	frappe.response["filename"] = f"{_(filename)}.{extension}"
+
+
+def send_report_email(user_email, file_name, file_extension, content, attached_to_name):
+	_file = frappe.get_doc(
+		{
+			"doctype": "File",
+			"file_name": f"{file_name}.{file_extension}",
+			"attached_to_doctype": "Report",
+			"attached_to_name": attached_to_name,
+			"content": content,
+			"is_private": 1,
+		}
+	)
+	_file.save(ignore_permissions=True)
+
+	file_url = _file.get_url()
+	file_url = frappe.utils.get_url(file_url)
+	file_retention_hours = frappe.get_system_settings("delete_background_exported_reports_after") or 48
+	frappe.sendmail(
+		recipients=[user_email],
+		subject=frappe._("Your exported report: {0}").format(file_name),
+		message=frappe._(
+			"The report you requested has been generated.<br><br>"
+			"Click here to download:<br>"
+			f"<a href='{file_url}'>{file_url}</a><br><br>"
+			f"This link will expire in {file_retention_hours} hours."
+		),
+	)
+
+
+def delete_old_exported_files():
+	file_retention_hours = frappe.get_system_settings("delete_background_exported_reports_after") or 48
+
+	cutoff = frappe.utils.add_to_date(frappe.utils.now_datetime(), hours=-file_retention_hours)
+	old_files = frappe.get_all(
+		"File",
+		filters={
+			"attached_to_doctype": "Report",
+			"creation": ("<", cutoff),
+		},
+		pluck="name",
+	)
```

Comment:
This will delete all the attachments of the report, regardless of if the user manually attached it.

Reply:
We need someway flag these exports 

Cc: @iamejaaz

Reply:
Set some specific filename and filter for that?